### PR TITLE
we should have training and test sets

### PR DESCRIPTION
**Pull request by XD-DENG**
I split the parsed data into two sets for training and testing separately. I guess the original author didn't consider this so that everyone can re-produce the result. But I think it would be better to have training set and test set, since this is a quite common concept in machine learning and beginners may get confused if we don't consider this.

### CODE REVIEW

**Comment by srowen** [comment]
This isn't quite how we report issues or suggest changes. Please read https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark first.

This is just an example of getting metrics, so the test/train split isn't as important. Still it is showing evaluating metrics and another example in isotonic_regression_example.py does do a test/train split. You might follow its example. Look for other similar examples that should be standardized in this way.

**Comment by XD-DENG** [comment]
Hi Owen, thanks for clarification. 

Sure, understood. 

I proposed this PR as I was a little bit confused myself given train/test split should be a fundamental step when doing regression. It's okay surely if you only set it for showing metrics evaluating.

**Comment by srowen** [comment]
@XD-DENG can you address my comments or close this PR?

**Comment by XD-DENG** [comment]
@srowen Hi Owen, sure. Thanks a lot for your clarification. 

My understanding was that you find this modification unnecessary, so I didn't proceed further. 

Happy new year.

**Comment by srowen** [comment]
OK, either way. I was suggesting that it's fine to make this example consistent with the others, even if test/train isn't the point of this example. if you do proceed, read the contributing wiki above to see how to make a JIRA, and then investigate whether there are other similar examples to change like the one I mentioned.

**Comment by XD-DENG** [comment]
Thanks for clarifying. Will have a look if I can proceed as you suggested with JIRA. 

Thanks
